TwoNote/app/src/main/java/com/microsoft/device/display/samples/twonote/NoteDetailFragment.kt

```diff
@@ -141,13 +184,33 @@ class NoteDetailFragment : Fragment() {
         val purpleButton = view.findViewById<Button>(R.id.button_purple)
         purpleButton.setOnClickListener { chooseColor(PaintColors.Purple.name) }
 
+        val chooseButton = view.findViewById<ImageButton>(R.id.button_choose)
+        chooseButton.setOnClickListener {
+            val textInput = TextInputEditText(requireContext())
+
+            AlertDialog.Builder(requireContext())
+                .setMessage(resources.getString(R.string.choose_color_message))
+                .setView(textInput)
+                .setCancelable(true)
+                .setPositiveButton(resources.getString(R.string.ok)) { dialog, _ ->
+                    val result = stringToColor(textInput.text.toString())
+                    if (result != -1) {
+                        chooseColor("", result)
+                    }
+                    dialog.dismiss()
+                }
+                .setTitle(resources.getString(R.string.choose_color))
```

Comment:
something for the future, we should add a cutsom color tile so the user can see what they chose before drawing with it. we can set the color of the tile using `tile.setColorFilter` or something along those lines

Reply:
ooh i can try to add that now!